### PR TITLE
init commit

### PR DESCRIPTION
**Pull request by eevans**
The Makefile here is cribbed from blubber
(https://phabricator.wikimedia.org/source/blubber/browse/master/Makefile),
and setup to cross-compile binaries for linux, windows and.  It uses a
locally overridden `GOPATH` (mwctl/.build) and so when you build, it
has to fetch and build anything that isn't already in `GOROOT`, which
includes the relevant parts of standard lib for non-native platforms.

TL;DR The non-native builds will take a bit longer to build.  You
should use the apropos make target (i.e. `make linux`) while iterating
development of mwctl, and then build all to check-in any changes.

### CODE REVIEW

**Comment by Pchelolo** [comment]
+1

File: .gitignore [hunk 1]
```diff
@@ -12,3 +11,9 @@
 
 # Project-local glide cache, RE: https://github.com/Masterminds/glide/issues/736
 .glide/
+
+# Emacs (yes, Emacs)
```
Comment (d00rman) [inline .gitignore:15]: grrr :)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	cd $(BUILD_DIR) && GOOS=windows GOARCH=$(GOARCH) $(GO) build -v -x -o mwctl-windows-$(GOARCH).exe
+	mv $(BUILD_DIR)/mwctl-windows-$(GOARCH).exe $(BINDIR)
+
+darwin: setup
+	# MacOS
+	cd $(BUILD_DIR) && GOOS=darwin GOARCH=$(GOARCH) $(GO) build -v -x -o mwctl-darwin-$(GOARCH)
+	mv $(BUILD_DIR)/mwctl-darwin-$(GOARCH) $(BINDIR)
+
+
+.PHONY: clean all
```
Comment (d00rman) [inline Makefile]: Missing `setup linux windows darwin`

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (d00rman) [inline Makefile:1]: What about `make install` ? I can see the usefulness of having `mwctl` in your path (once things are stabilised, ofc)

Comment (d00rman) [reply Makefile:1]: Also, should `bin/mwctl` be made a symlink to the platform-dependent binary?

File: config.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	"encoding/json"
+)
+
+type Config struct {
+	Sources map[string]string `json:"sources"`
+}
+
+func ReadConfig(data []byte) (*Config, error) {
+	var config Config
+	err := json.Unmarshal(data, &config)
```
Comment (d00rman) [inline config.go]: YAML, YAML pleaseeee :)

**Comment by d00rman** [comment]
Some minor nits, but the first stub looks :+1:

File: Makefile (hunk 2 above)
Comment (eevans) [reply Makefile]: Yup.

File: Makefile (hunk 3 above)
Comment (eevans) [reply Makefile:1]: Any suggestions for how to approach this cleanly in a multi-platform manner?

File: config.go (hunk 4 above)
Comment (eevans) [reply config.go]: I wasn't going to go there because it requires an external dependency, but it occurs to me that we'll need YAML handling anyway, so... yeah, OK. :)

File: config.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"github.com/mitchellh/go-homedir"
```
Comment (Pchelolo) [inline config.go:4]: Ok, now that we're bringing in dependencies, should we use vending now? https://blog.gopheracademy.com/advent-2015/vendor-folder/

File: Makefile (hunk 3 above)
Comment (d00rman) [reply Makefile:1]: Yeah, not a small feat, let's leave it for next time :)

File: config.go (hunk 5 above)
Comment (d00rman) [reply config.go:4]: The Makefile uses `.build/` with the exact same purpose, so yeah, we should rename it probably.

File: Makefile (hunk 3 above)
Comment (d00rman) [reply Makefile:1]: Actually, having the binaries checked in the repo creates _binary noise_. What about ignoring them and instructing the build system to build only the binary for the system being built on? We can always keep the rules for individual platforms, so as to be able to build them on demand (for packaging purposes, for example).

Comment (eevans) [reply Makefile:1]: This is possible, but adds the requirement that people have a working build environment.  My suggestion was going to be that we publish binaries as Github releases, but I had planned to refrain from suggesting that until the dust had settled and we (at least) had something worthy of calling a release.

Comment (d00rman) [reply Makefile:1]: Yup, I think GH releases should be the way to go
